tools/prepare_release.csx

```diff
@@ -57,31 +67,85 @@ public static string ReadGitHubToken()
+        .Single(release => release.Name == existingReleaseName && release.Draft == true);
     Console.WriteLine($"Fetched GitHub release '{existingRelease.Name}'");
     return existingRelease;
 }
 
-public static async Task<Issue> GetExistingIssue(Milestone existingMilestone)
+public static async Task<IList<Issue>> GetIssuesInMilestone(Milestone milestone)
+{
+    Console.WriteLine($"Fetching issues in milestone '{milestone.Title}'...'");
+    var issueRequest = new RepositoryIssueRequest { Milestone = milestone.Number.ToString(), State = ItemStateFilter.All };
+    var issues = (await gitHubClient.Issue.GetAllForRepository(repoOwner, repoName, issueRequest)).ToList();
+    Console.WriteLine($"Fetched {issues.Count} issues in milestone '{milestone.Title}'");
+    return issues;
+}
+
+public static IEnumerable<string> GetIssuesReferencedFromRelease(Release release)
+{
+    var issuesReferencedFromRelease = new HashSet<string>();
+    foreach (Match match in Regex.Matches(release.Body, @"\(#([0-9]+)\)"))
+    {
+        issuesReferencedFromRelease.Add(match.Groups[1].Value);
+    }
+    return issuesReferencedFromRelease;
+}
+
+public static Issue PopExistingReleaseIssue(IList<Issue> issues)
 {
-    Console.WriteLine("Fetching release issue 'Release vNext'...");
-    var issueRequest = new RepositoryIssueRequest { Milestone = existingMilestone.Number.ToString(), State = ItemStateFilter.Open };
-    var existingIssue = (await gitHubClient.Issue.GetAllForRepository(repoOwner, repoName, issueRequest))
-        .Single(issue => issue.Title == "Release vNext");
-    Console.WriteLine($"Fetched release issue #{existingIssue.Number}: '{existingIssue.Title}'");
-    return existingIssue;
+    var issue = issues.Single(i => i.Title == "Release " + existingReleaseName);
+    Console.WriteLine($"Found release issue #{issue.Number}: '{issue.Title}'");
+    issues.Remove(issue);
+    return issue;
+}
+
+public static void CrossReferenceIssues(IEnumerable<Issue> issuesInMilestone, IEnumerable<string> issueNumbersReferencedFromRelease)
+{
+    var issueNumbersInMilestone = issuesInMilestone.Select(i => i.Number.ToString());
+    var issueNumbersInReleaseButNotMilestone = issueNumbersReferencedFromRelease.Except(issueNumbersInMilestone);
+    var issuesInMilestoneButNotRelease = issuesInMilestone.Where(i => !issueNumbersReferencedFromRelease.Contains(i.Number.ToString()));
+
+    if (!issuesInMilestoneButNotRelease.Any() && !issueNumbersInReleaseButNotMilestone.Any())
+    {
+        Console.WriteLine("The release refers to the same issues included in the milestone. Congratulations.");
+        return;
+    }
+
+    Console.WriteLine();
+
+    if (issuesInMilestoneButNotRelease.Any())
+    {
+        Console.WriteLine("The following issues are linked to the milestone but not referenced in the release:");
+        foreach (var issue in issuesInMilestoneButNotRelease)
+        {
+            Console.WriteLine($"  #{issue.Number} - {issue.Title}");

```

Comment:
Printing title here because we have it. For issues referenced from the release, we don't have a title. We _could_ look it up, but I wasn't sure there was enough benefit.